### PR TITLE
Add GitHub action to build examples

### PR DESCRIPTION
**Pull request by VladimirUmek**

### CODE REVIEW

File: .github/workflows/examples.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+jobs:
+  Build:
+    strategy:
+      fail-fast: false
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout NXP_IMXRT1050-EVKB_BSP
```
Comment (jkrech) [inline .github/workflows/examples.yml:24]: I am not an expert in actions, but wouldn't it be great if we would not need to customize these actions and could reuse them.
We could search for *.pdsc file (there should be only one!)  to register the local pack
We could search for csolution.yml files to run all examples 
etc.

Comment (RobertRostohar) [reply .github/workflows/examples.yml:24]: All of that is possible but would probably increase the complexity of the action. Currently the action is fairly simple and easy to read and understand. That might change if we add more automation in order to reduce customization.

Comment (VladimirUmek) [reply .github/workflows/examples.yml:24]: We could do this for repositories with a simple structure - like this one. Current actions script is simple as possible and as soon as we do searching etc., the action will become bloated with additional commands and processing.
I'm not sure that on the long run all repositories will stay as simple as few ones that we currently have and that we will be able to scale a generic action script...

Comment (RobertRostohar) [reply .github/workflows/examples.yml:24]: I would merge the action as it for now so that it gets used. We can still consider changes later.

**Pull request merged**